### DISCUSSION TITLE
JSON field changes

### DISCUSSION THREAD

**Discussion by andrewpitts15** [spatie/laravel-activitylog#994] [Q&A]
On my model, I have a JSON field called "answers" that contains all the answers given by the user for various forms that can each have different questions based on the form schema. 

```
class FormResponse extends Model
{
    use HasFactory;
    use LogsActivity;

    public $guarded = [];

    protected $casts = [
        'answers' => 'array',
    ];

    public function getActivityLogOptions(): LogOptions
    {
        return LogOptions::defaults()
            ->logOnly(['answers'])
            ->logOnlyDirty()
            ->dontLogIfAttributesChangedOnly(['updated_at']);
    }
}
```

I would like any changes to response to be stored in the activity log, but exclude any unchanged fields from the change log. The code I have is unfortunately storing values that haven't changed as well as those that have.

This is what I'm getting:

```
{"old": {"answers": {"name": "foo", "occupation": "Programmer"}}, "attributes": {"answers": {"name": "foo bar", "occupation": "Programmer"}}}
```

This is what I'd like
```
{"old": {"answers": {"name": "foo"}}, "attributes": {"answers": {"name": "foo bar"}}}
```
Is this possible? If so how do I do this? Thanks.

**Comment by Gummibeer** [comment]
Hey,
this is a perfect use-case for the new v4 changes pipeline:
https://spatie.be/docs/laravel-activitylog/v4/advanced-usage/manipulate-changes-array
You can create your own pipe that manipulates the changes array how you want it.
